sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaIO.java

```diff
@@ -844,6 +858,11 @@ public PartitionState apply(TopicPartition tp) {
           partitionStates.get(i).nextOffset = ckptMark.getNextOffset();
         }
       }
+
+      messagesConsumedPerSplit = Metrics.counter("io.splits", source.id + ".messagesConsumed");
+      bytesConsumedPerSplit = Metrics.counter("io.splits", source.id + ".bytesConsumed");
+      backlogBytesPerSplit = Metrics.gauge("io.splits", source.id + ".backlog.bytes");
+      backlogMessagesPerSplit = Metrics.gauge("io.splits", source.id + ".backlog.messages");
```

Comment:
I agree these would be informative, however they execute in a separate thread and do not work well with the way metrics are designed.
You could try discussing this with @bjchambers to see if there is something we can do about this.